File: solutions/alfi.compitest/source_gen/alfi/compitest/EnumTrafficLightColorToString.cpp
#include "EnumTrafficLightColorToString.h"
#include <string>
#include "EnumTrafficLightColor.h"

std::string EnumTrafficLightColorToString(EnumTrafficLightColor value)
{
  if (value == EnumTrafficLightColor::RED)
  {
    return std::string("RED");
  }

  if (value == EnumTrafficLightColor::GREEN)
  {
    return std::string("GREEN");
  }

  if (value == EnumTrafficLightColor::YELLOW)
  {
    return std::string("YELLOW");
  }

  return std::string("?");
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/EnumTrafficLightColorToString.cs
/// <filename>
///     EnumTrafficLightColorToString.cs
/// </filename>
public class EnumTrafficLightColorToString
{
    public static string EnumTrafficLightColorToString(EnumTrafficLightColor value)
    {
        if (value == EnumTrafficLightColor.RED)
        {
            return "RED";
        }

        if (value == EnumTrafficLightColor.GREEN)
        {
            return "GREEN";
        }

        if (value == EnumTrafficLightColor.YELLOW)
        {
            return "YELLOW";
        }

        return "?";
    }
}
